### ISSUE TITLE
Make `authorize` more lenient

### ISSUE THREAD

**Issue by KtorZ** [truqu/elm-oauth2#6]
Echoing #5 and #4, it would come in handy to be able to specify some custom parameters for the query of the `authorize` function. This could be achieved via a:

```elm
authorizeWithOpts :: (QueryString -> QueryString) -> Authorize -> Command msg
```

There's also probably some work that can be done to leverage the `parse` functions from the Implicit and AuthorizationCode modules.

**Comment by roman-mkh** [comment]
Hello @KtorZ
I'm missing such advanced function too, Our corp auth server requires additional non standard query parameters for the implicit grant type. 
I think you mix up the signature with haskell :smile:. IMHO it should look like
`authorizeWithOpts : (QueryString -> QueryString) -> Authorization -> Cmd msg`

**Comment by KtorZ** [comment]
Solved in 4.0.0
